File: Labs/Lab 2/Lab2.java
public class Lab2 {

    public static boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) {
            return false;
        }

        int[] counts = new int[256];

        for (int i = 0; i < s.length(); i++) {
            counts[s.charAt(i)]++;
            counts[t.charAt(i)]--;
        }

        for (int count : counts) {
            if (count != 0) {
                return false;
            }
        }

        return true;
    }

    public static void main(String[] args) {
        String s = "listen";
        String t = "silent";
        System.out.println(isAnagram(s, t));

        s = "rat";
        t = "car";
        System.out.println(isAnagram(s, t));
    }
}

File: Labs/Lab 2/Lab2.py
class Lab2:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        counts = {}

        for char_s, char_t in zip(s, t):
            counts[char_s] = counts.get(char_s, 0) + 1
            counts[char_t] = counts.get(char_t, 0) - 1

        for count in counts.values():
            if count != 0:
                return False

        return True


if __name__ == "__main__":
    solution = Solution()
    s = "listen"
    t = "silent"
    print(solution.isAnagram(s, t))

    s = "rat"
    t = "car"
    print(solution.isAnagram(s, t))
